common/persistence/serialization/task_serializer.go

```diff
@@ -240,6 +240,50 @@ func (s *TaskSerializer) DeserializeVisibilityTasks(
 	return taskSlice, nil
 }
 
+func (s *TaskSerializer) SerializeTieredStorageTasks(
+	taskSlice []tasks.Task,
+) (map[tasks.Key]commonpb.DataBlob, error) {
+	blobSlice := make(map[tasks.Key]commonpb.DataBlob, len(taskSlice))
+	for _, task := range taskSlice {
+		var tieredStorageTask *persistencespb.TieredStorageTaskInfo
+		switch task := task.(type) {
+		case *tasks.TieredStorageTask:
```

Comment:
`TieredStorageTask` -> maybe something less generic?

Reply:
I thought about it but didn't come up with a better name. Another candidate was `ArchivalTask`. Do you have any suggestions?

Reply:
i am not good at naming ... 
we can change the type name later, not a big deal